tephra-core/src/main/java/org/apache/tephra/persist/AbstractTransactionLog.java

```diff
@@ -211,26 +212,36 @@ private void sync() throws IOException {
       // prevent writer being dereferenced
```

Comment:
I am not sure why this is needed. As long as this method is running, there is a stack frame (= a GC root) that has a reference to this, and hence to its members. Anybody know why this is here?

Reply:
I don't think it is needed since writer only get assigned once in the `init` method. It was there since the beginning without any reason.

Reply:
That's what i thought... removing it now.